components/clarity-repl/src/analysis/ast_visitor.rs

```diff
@@ -144,14 +144,17 @@ pub trait ASTVisitor<'a> {
                             args.get(1).unwrap_or(&DEFAULT_EXPR),
                             args.get(2).unwrap_or(&DEFAULT_EXPR),
                         ),
-                        DefineFunctions::Trait => self.traverse_define_trait(
-                            expr,
-                            args.get(0)
-                                .unwrap_or(&DEFAULT_EXPR)
-                                .match_atom()
-                                .unwrap_or(&DEFAULT_NAME),
-                            &args[1..],
-                        ),
+                        DefineFunctions::Trait => {
+                            let params = if args.len() >= 1 { &args[1..] } else { &[] };
```

Comment:
I'm surprised that rust lets us write unsafe code like that (`args[1..]` even if args length can be 0). 
I used this `if else` pattern in multiple places, let me know if there's a better way to do so

Reply:
That works!